### ISSUE TITLE
Adding Angular Material v2.0 or Ionic 2.0  to the Advanced Seed

### ISSUE THREAD

**Issue by OscarAgreda** [NathanWalker/angular-seed-advanced#37]
for the past 3 months, I have spent hours, even full whole days doing hard core Mad Science trying to decide between .
-Xamarin - i am a senior .Net developer., my days are full of C#E, besides Xaml comes naturally
-Apache Cordova , with 3 years with Angular and built 3 enterprise apps, i now love my JavaScript and angular coding ... love it so much , so i wanted to hybrid with Cordova. unfortunately plug-in are a pain to work with and specially maintain consistency , besides Hybrid are not Native..
-React Native , well everybody was telling me to go this route 

but none of the above 3 convinced me  or my team.  
until I saw the angular 2 advanced seed. by far the best contribution someone can do for all developers,

By creating this template you have truly opened the doors and designed a true path towards Angular 2.0 as well as started a revolution on how software applications will be from now on.  

In my humble opinion , i think you will make it perfect  by adding Ionic 2 or Angular Material 2 to the mix.
is that on you radar ?

**Comment by NathanWalker** [comment]
Hi @OscarAgreda, this is really great to hear, thank you and you're welcome.
Many thanks must also go to @mgechev and the many contributors on the fantastic parent seed [angular2-seed](https://github.com/mgechev/angular2-seed).

You may be happy to hear a fantastic contributor, @d3viant0ne, has been working on `Angular Material 2` integration here: https://github.com/DeviantJS/angular2-seed-material2-advanced/pull/2
And we will likely merge that in with master here, so **yes** to plans to integrate with Material...soon.

No plans to integrate with Ionic 2 at moment, however I have been considering adding a `hybrid` framework which would provide a `cordova` option if a team desired. This `cordova` integration could be done with Ionic 2. I personally would not be able to provide that support for several months but eventually maybe.

Again, thank you for your kind words and feel free to drop feedback/suggestions on anything as new issues. The goal is to provide the best out of box experience for large projects; trying to maintain a balance between `not enough` to `too much` can be delicate with a seed, so keep that in mind as far as requesting `core` features. :+1:

**Comment by joshwiens** [comment]
@NathanWalker @OscarAgreda I was actually pondering an Ionic2 integration this weekend if for nothing else, covering most of the major integrations.

On a personal note, i'm currently vetting hybrid-mobile offerings for work. Ionic2 is on that list along with NativeScript. Lucky @NathanWalker did a kick ass job on this one which only leaves Ionic2.

I'm pretty set on the integration & i'm happy to make it public. For the sake of consistency, i'll keep it inline with Nathan's Advanced seed to simplify comparison for people needing to make the same choice I do.

On an angular2-material note - They just tagged alpha.2 which is what I was waiting on.
Expect to see a Material2 version of the Advanced seed Sunday(ish).

**Comment by joshwiens** [comment]
@OscarAgreda - Some things to note with @angular2-material

> I'm adding it to a fork of this repo because it's simply not ready to be in the parent
> - Plethora of console errors
> - Style issues
> - Limited arrangement options ( very happy path layout wise )
> - Limited usability
> - Issues with packaging / lazy loading

@NathanWalker is a fan of Material2 as am I, when it's in a state that would not be detrimental to the user base as a whole, what's in the Material2 fork will most likely be rolled down to this repo and the fork deleted. I would expect that to happen at some point in the Material2 Beta timeframe assuming @NathanWalker wants Material2 in the advanced-seed#master ( pretty sure I know the answer there ).

I'm currently updating the Material2 fork of the basic seed, once i've taken that as far as possible given the limitations, i'll roll that up and start tagging versions of Material2 in the forks.

**Comment by mtabaj** [comment]
Hi Oscar,
What is the situation with Material2?  May we expect to be in the project soon?

**Comment by joshwiens** [comment]
@mtabaj Material2 is an Alpha, that should say everything you need to know about it.

As for seeing it in the Advanced Seed, list I have heard from @NathanWalker the approach will be to integrate common libraries in branches or forks but not into seed#master as it limits the scope of the users that can use the project without gutting libraries they don't need.

As far as adding Material2 to the advanced-seed, at it's core it's no different than the seed it is extended from and there is a fork of the parent seed implementing Material2. If you want to add it to your fork of the Advanced Seed, that would be a good place to start.

**Comment by joshwiens** [comment]
@mtabaj I'll be adding the Material2 implementation in the branch below this weekend. Hoping they cut alpha5 on Friday which is what I am waiting on ( it uses rc1 not rc0 )

https://github.com/NathanWalker/angular2-seed-advanced/tree/material2

**Comment by joshwiens** [comment]
FYI - They tagged Alpha5 last night, i'll be moving ahead this weekend as planned.

**Comment by joshwiens** [comment]
@NathanWalker - One question on this "Material" thing I am on the fence about.
- The pure implementation is to simply add Material2 on the angular side, document & call it a day.
- The complete solution would be to maintain a consistent UI and materialize the {N} app.

I know the right answer there, problem is that I don't like the "right" answer. The pure implementation's inconsistent look and feel bugs me as it's not how I would chose to roll in any project.

It would be a bit more work but calling the branch something like material-design and then implementing that with both Material2 & {N} plugins to provide a complete solution may be of more value as it's closer to what would be done on actual projects.

Your project, your call.

**Comment by OscarAgreda** [comment]
Hello @d3viantOne
Any news on your last comment ?

**Comment by joshwiens** [comment]
@OscarAgreda It's next on my list after I finish with the parent seed material2 fork ( doing that right now )
Everything was on hold for a refactoring that ended up not being worth doing upstream.

I should have it done this evening ( depending on testing issues with material2 / zone ). Documentation on `how` to add it should realistically be done tomorrow.

**Comment by joshwiens** [comment]
@OscarAgreda - FYI there are still issues with Material2 / Testing without deviating from the parent seed.

Pushing the changes to the material2 branch and it may or may not be PR'd depending on the changes required to get the testing working. Nathan took a different approach to testing which may or may not work out in our favor in the advanced seed.

**Comment by OscarAgreda** [comment]
I am sure you are aware of this , but just in case 

the angular-material core-team is now stopping development on Material 1 
https://groups.google.com/forum/#!topic/ngmaterial/4ActiQp3nA0

Which means they will totally increase the rate of Material2 development .

**Comment by joshwiens** [comment]
@OscarAgreda - If I know Jeremy, he will hold the steady pace and maintain quality and control of Material2.

Material1 is a hot mess as a project, I'll be you lunch he keeps the reigns tight on Material2.

**Comment by ernestlombardi** [comment]
@NathanWalker @OscarAgreda Would love to use Material2 in my project. Any ETA on a working branch for angular2-seed-advanced?

**Comment by shlomiassaf** [comment]
@d3viant0ne Some of the packages in `material2` are web platform specific. 
This mainly related to the **Overlay** and **Portal** components in `material2/core`.

The 2 are the base foundation for a lot of components to come, such as drop down, dialogs and every component that needs an overlay.

I know Jeremy doesn't plan to support multiple platforms.

What are the plans for this integration, in some scenarios you can replace the implementation via DI (e.g: replace `OverlayContainer` or `DomPortalHost`) but in others it might be difficult (`ConnectedPositionStrategy`)...

In general these overlay's are handled quite differently  in mobile vs web. Having Overlay/Portal a base material 2 concept how are you going to handle that?

**Comment by hungtruongquoc** [comment]
Hi all, is there any update on the integration with Material 2 so far? Or is there any specific guide lines to manually integrate to this seed css framework like Bootstrap? Thanks.

**Comment by NathanWalker** [comment]
Closing as there are no plans to integrate either of these in this seed at moment. Best to handle in forks etc.
